third_party/amd/lib/TritonAMDGPUTransforms/StreamPipelineV2.cpp

```diff
@@ -207,9 +207,21 @@ getSharedEncIfAllUsersAreDotEnc(Value val) {
       auto CTALayout = ttg::getCTALayout(srcTy.getEncoding());
       auto order = ttg::getOrder(srcTy.getEncoding());
       unsigned bitWidth = srcTy.getElementType().getIntOrFloatBitWidth();
+      SmallVector<unsigned> sharedOrder;
+      int rank = order.size();
+      // TODO rework this when shared -> dotOp conversions support arbitrary
```

Comment:
Nit: "TODO: rework this when shared -> dotOperand conversions support 3D cases."

Reply:
Not sure I got this.

Specific order in shared layout is required, because MMA **shared->dot op conversions** (both AMD and NV) support only this case for now.

3D tensors are already supported, but batch dimension is required to be the slowest one. I.e. order field of shared layout required to look like this `[?, ?, 0]`.

Reply:
Oh it's more about shared to dotOperand. we have dotOperand layout there, not dotOp layout. It's a small issue.